waspc/src/Wasp/AppSpec/Entity.hs

```diff
@@ -45,5 +47,15 @@ getPslModelBody = pslModelBody
 getIdField :: Entity -> Maybe PslModel.Field
 getIdField = findIdField . getPslModelBody
 
+doesFieldHaveAttribute :: String -> String -> Entity -> Maybe Bool
+doesFieldHaveAttribute fieldName attrName entity =
+  doesPslFieldHaveAttribute attrName <$> findPslFieldByName fieldName entity
+
+findPslFieldByName :: String -> Entity -> Maybe PslModel.Field
+findPslFieldByName fieldName Entity {pslModelBody = PslModel.Body elements} =
+  find isField [field | (PslModel.ElementField field) <- elements]
+  where
+    isField PslModel.Field {_name = name} = name == fieldName
+
```

Comment:
@infomiho @Martinsos 

I wasn't sure how to decide what to put in here vs. in `Psl/Util.hs`.

Also, this is the only file that uses `Psl/Util.hs`, should we maybe remove it (especially considering the conversation about keeping `PslModel`  stuff hidden)?

Reply:
I am missing a bit of context so it is somewhat hard for me to make a decisive statement here, but:
1. If we have functions operating on PSL stuff and not knowing anything about Entity, then it sounds good to have them in PslUtil.hs .

For future, it might help if you give more context. For exmaple, what is in Psl/Util.hs, why do you think it shouldn't be there (besides this being the only place using it), what are pros/cons of each approach (aka "what have you tried"), what do getIdField and getIdBlockAttribute do to make them candidates for this refactoring.

Reply:
Also, I like the idea of exposing Entity + ENtity.Filed interface and hiding PSL if possible. If not possible / practical, we can let some PSL seep out.
I can't judge right now how possible / practical it is without diving deeper, but I believe you will know that better!

Reply:
I liked the split of having `Entity` using helper functions from `Psl/Util.hs`.

Example of such fn:
```hs
getIdField :: Entity -> Maybe PslModel.Field
getIdField = findIdField . getPslModelBody
```

All the low level work on the PSL schema is hidden away in the `PslUtil` module, while Entity is composing those function to get out what they need without worrying about the implementation. 

For this PR, I think the split could go like this:
1. `Psl/Util.hs` implements functions for checking if certain field in PSL has some attribute attached `isAttributePresentOnField` 
2. `Entity.hs` uses that function to come up with a function `isFieldUnique` (looking for `@unique`)
3. `Valid.hs` can then do something like `Entity.isFieldUnique "username"`

Reply:
The main idea was as follows:

If we're hiding PSL as an implementation detail (this is the core assumption), there probably aren't many reasons for having a `Utils` file (in its current location).

The only module what should ever be using those utils is `Entity` (since all PSL stuff is a "hidden" implementation detail), so we might as well move all those functions there (instead of having them globally available from a different part of the codebase).

Such an organisation colocates all relevant code and discourages/prevents us from acting on PSL directly.

Reply:
I've added the `isFieldUnique` function and left the other stuff as is for now.

Reply:
@sodic not so important for this PR, but I would do the opposite -> I would put PSL specific stuff in PSL/Utils.hs or something like that, not in `Entity`, IF it makes sense to be standalone. This is useful because it clearly indicates that this code is not Entity specific, in the sense that there are some assumptions they share. Entity would still be having that as an implementation details, that doesn't change. And the fact that nobody else is using it is also not an issue, if it is general it is general -> we might need it in some other parts of the code, even if also hidden as an implementation detail.
Otherwise we could just move the whole PSL module under Entity. But I like it better that we say "hey, look we have PSL module because Wasp knows how to work with it!" without assuming who is using it.

But this is not a biggie, we can in any case move it around if needed.